### PR TITLE
Refactor FabricateCandySystem Into Its Generic Equivalent

### PR DESCRIPTION
**Pull request by Mnemotechnician**
# Description
Refactors the nyano shitcode responsible for allowing borgs to dispense candies into a more generic variant, that allows to define what action dispenses what entity, and allowing an arbitrary number of such actions on an entity.

Requested by @DangerRevolution 

<details><summary><h1>Media</h1></summary><p>

https://github.com/user-attachments/assets/b4d643e2-c9b0-4367-8b9c-2d0cd4a228b9

</p></details>

# Changelog
No cl no fun

### CODE REVIEW

**Review by DangerRevolution — APPROVED** [review]
W rizZ

File: Resources/Prototypes/Nyanotrasen/Actions/types.yml [hunk 1]
```diff
@@ -165,7 +165,9 @@
   - type: InstantAction
     icon: { sprite: Nyanotrasen/Objects/Consumable/Food/candy.rsi, state: lollipop }
     useDelay: 120
-    event: !type:FabricateLollipopActionEvent
+    event:
+      !type:FabricateActionEvent
+      fabrication: FoodLollipop
```
Comment (VMSolidus) [inline Resources/Prototypes/Nyanotrasen/Actions/types.yml:170]: Move these actions to Prototypes/Actions/types.yml as well.

**Review by VMSolidus — CHANGES_REQUESTED** [review]
I'll approve this once the two borg actions are moved out of Nyano. :)

File: Resources/Prototypes/Nyanotrasen/Actions/types.yml (hunk 1 above)
Comment (Mnemotechnician) [reply Resources/Prototypes/Nyanotrasen/Actions/types.yml:170]: within 3-5 business days

File: Content.Server/Abilities/Borgs/FabricateActionsSystem.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+using Content.Shared.Actions;
+using Content.Shared.Actions.Events;
+
+namespace Content.Server.Abilities.Borgs;
+
+public sealed partial class FabricateActionsSystem : EntitySystem
+{
+    [Dependency] private readonly ActionBlockerSystem _actionBlocker = default!;
+    [Dependency] private readonly SharedActionsSystem _actions = default!;
+
```
Comment (DEATHB4DEFEAT) [inline Content.Server/Abilities/Borgs/FabricateActionsSystem.cs:11]:
```suggestion


```

File: Content.Server/Abilities/Borgs/FabricateActionsSystem.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    public override void Initialize()
+    {
+        base.Initialize();
```
Comment (DEATHB4DEFEAT) [inline Content.Server/Abilities/Borgs/FabricateActionsSystem.cs:14]:
```suggestion
        base.Initialize();

```

File: Content.Server/Abilities/Borgs/FabricateActionsSystem.cs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        SubscribeLocalEvent<FabricateActionsComponent, ComponentStartup>(OnStartup);
+        SubscribeLocalEvent<FabricateActionsComponent, ComponentShutdown>(OnShutdown);
+        SubscribeLocalEvent<FabricateActionsComponent, FabricateActionEvent>(OnFabricate);
+    }
+
```
Comment (DEATHB4DEFEAT) [inline Content.Server/Abilities/Borgs/FabricateActionsSystem.cs:19]:
```suggestion


```

File: Content.Shared/Actions/Events/FabricateActionEvent.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Actions.Events;
+
+public sealed partial class FabricateActionEvent : InstantActionEvent
```
Comment (DEATHB4DEFEAT) [inline Content.Shared/Actions/Events/FabricateActionEvent.cs:5]:
```suggestion
public sealed class FabricateActionEvent : InstantActionEvent
```

**Review by DEATHB4DEFEAT — APPROVED** [review]

**Comment by DangerRevolution** [comment]
Because the changes are just minor line changes; I'm going to merge the suggestions and then merge the PR :)

File: Content.Shared/Actions/Events/FabricateActionEvent.cs (hunk 5 above)
Comment (Mnemotechnician) [reply Content.Shared/Actions/Events/FabricateActionEvent.cs:5]: It's a DataDefinition so it needs to be partial

Comment (Mnemotechnician) [reply Content.Shared/Actions/Events/FabricateActionEvent.cs]: Fuck, I accidentally applied the suggestion too... well, fuck me.

Comment (DEATHB4DEFEAT) [reply Content.Shared/Actions/Events/FabricateActionEvent.cs]: webed**it**:tm:

**Comment by DangerRevolution** [comment]
going to re-run this until no more unrelated test fails

**Pull request merged**